From 21da849e373d237671645708a600f11bdb651819 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 20 Mar 2019 12:45:10 +0800
Subject: [PATCH 606/706] crypto: caam: remove unused codes and fix header
 files

It seems dpseci_set_opr is not used by anyone, so we remove it and fix
the wrong header file references.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/crypto/caam/caamalg_qi2.c |  4 +-
 drivers/crypto/caam/caamalg_qi2.h |  4 +-
 drivers/crypto/caam/dpseci.c      | 85 -------------------------------
 3 files changed, 4 insertions(+), 89 deletions(-)

diff --git a/drivers/crypto/caam/caamalg_qi2.c b/drivers/crypto/caam/caamalg_qi2.c
index 454ad667ebc9..0686272006d1 100644
--- a/drivers/crypto/caam/caamalg_qi2.c
+++ b/drivers/crypto/caam/caamalg_qi2.c
@@ -44,8 +44,8 @@
 #include "key_gen.h"
 #include "caamalg_desc.h"
 #include "caamhash_desc.h"
-#include "../../../drivers/staging/fsl-mc/include/dpaa2-io.h"
-#include "../../../drivers/staging/fsl-mc/include/dpaa2-fd.h"
+#include <soc/fsl/dpaa2-io.h>
+#include <soc/fsl/dpaa2-fd.h>
 
 #define CAAM_CRA_PRIORITY	2000
 
diff --git a/drivers/crypto/caam/caamalg_qi2.h b/drivers/crypto/caam/caamalg_qi2.h
index 2f21bcdca49c..c73620eb3a5e 100644
--- a/drivers/crypto/caam/caamalg_qi2.h
+++ b/drivers/crypto/caam/caamalg_qi2.h
@@ -35,8 +35,8 @@
 #ifndef _CAAMALG_QI2_H_
 #define _CAAMALG_QI2_H_
 
-#include "../../../drivers/staging/fsl-mc/include/dpaa2-io.h"
-#include "../../../drivers/staging/fsl-mc/include/dpaa2-fd.h"
+#include <soc/fsl/dpaa2-io.h>
+#include <soc/fsl/dpaa2-fd.h>
 #include <linux/threads.h>
 #include "dpseci.h"
 #include "desc_constr.h"
diff --git a/drivers/crypto/caam/dpseci.c b/drivers/crypto/caam/dpseci.c
index 7b7ea203b54d..ef357c1e7f6e 100644
--- a/drivers/crypto/caam/dpseci.c
+++ b/drivers/crypto/caam/dpseci.c
@@ -33,7 +33,6 @@
  */
 
 #include <linux/fsl/mc.h>
-#include "../../../drivers/staging/fsl-mc/include/dpopr.h"
 #include "dpseci.h"
 #include "dpseci_cmd.h"
 
@@ -698,90 +697,6 @@ int dpseci_get_api_version(struct fsl_mc_io *mc_io, u32 cmd_flags,
 	return 0;
 }
 
-/**
- * dpseci_set_opr() - Set Order Restoration configuration
- * @mc_io:	Pointer to MC portal's I/O object
- * @cmd_flags:	Command flags; one or more of 'MC_CMD_FLAG_'
- * @token:	Token of DPSECI object
- * @index:	The queue index
- * @options:	Configuration mode options; can be OPR_OPT_CREATE or
- *		OPR_OPT_RETIRE
- * @cfg:	Configuration options for the OPR
- *
- * Return:	'0' on success, error code otherwise
- */
-int dpseci_set_opr(struct fsl_mc_io *mc_io, u32 cmd_flags, u16 token, u8 index,
-		   u8 options, struct opr_cfg *cfg)
-{
-	struct fsl_mc_command cmd = { 0 };
-	struct dpseci_cmd_opr *cmd_params;
-
-	cmd.header = mc_encode_cmd_header(
-			DPSECI_CMDID_SET_OPR,
-			cmd_flags,
-			token);
-	cmd_params = (struct dpseci_cmd_opr *)cmd.params;
-	cmd_params->index = index;
-	cmd_params->options = options;
-	cmd_params->oloe = cfg->oloe;
-	cmd_params->oeane = cfg->oeane;
-	cmd_params->olws = cfg->olws;
-	cmd_params->oa = cfg->oa;
-	cmd_params->oprrws = cfg->oprrws;
-
-	return mc_send_command(mc_io, &cmd);
-}
-
-/**
- * dpseci_get_opr() - Retrieve Order Restoration config and query
- * @mc_io:	Pointer to MC portal's I/O object
- * @cmd_flags:	Command flags; one or more of 'MC_CMD_FLAG_'
- * @token:	Token of DPSECI object
- * @index:	The queue index
- * @cfg:	Returned OPR configuration
- * @qry:	Returned OPR query
- *
- * Return:	'0' on success, error code otherwise
- */
-int dpseci_get_opr(struct fsl_mc_io *mc_io, u32 cmd_flags, u16 token, u8 index,
-		   struct opr_cfg *cfg, struct opr_qry *qry)
-{
-	struct fsl_mc_command cmd = { 0 };
-	struct dpseci_cmd_opr *cmd_params;
-	struct dpseci_rsp_get_opr *rsp_params;
-	int err;
-
-	cmd.header = mc_encode_cmd_header(DPSECI_CMDID_GET_OPR,
-					  cmd_flags,
-					  token);
-	cmd_params = (struct dpseci_cmd_opr *)cmd.params;
-	cmd_params->index = index;
-	err = mc_send_command(mc_io, &cmd);
-	if (err)
-		return err;
-
-	rsp_params = (struct dpseci_rsp_get_opr *)cmd.params;
-	qry->rip = dpseci_get_field(rsp_params->rip_enable, OPR_RIP);
-	qry->enable = dpseci_get_field(rsp_params->rip_enable, OPR_ENABLE);
-	cfg->oloe = rsp_params->oloe;
-	cfg->oeane = rsp_params->oeane;
-	cfg->olws = rsp_params->olws;
-	cfg->oa = rsp_params->oa;
-	cfg->oprrws = rsp_params->oprrws;
-	qry->nesn = le16_to_cpu(rsp_params->nesn);
-	qry->ndsn = le16_to_cpu(rsp_params->ndsn);
-	qry->ea_tseq = le16_to_cpu(rsp_params->ea_tseq);
-	qry->tseq_nlis = dpseci_get_field(rsp_params->tseq_nlis, OPR_TSEQ_NLIS);
-	qry->ea_hseq = le16_to_cpu(rsp_params->ea_hseq);
-	qry->hseq_nlis = dpseci_get_field(rsp_params->tseq_nlis, OPR_HSEQ_NLIS);
-	qry->ea_hptr = le16_to_cpu(rsp_params->ea_hptr);
-	qry->ea_tptr = le16_to_cpu(rsp_params->ea_tptr);
-	qry->opr_vid = le16_to_cpu(rsp_params->opr_vid);
-	qry->opr_id = le16_to_cpu(rsp_params->opr_id);
-
-	return 0;
-}
-
 /**
  * dpseci_set_congestion_notification() - Set congestion group
  *	notification configuration
-- 
2.17.1

